### PR TITLE
changes for admin dashbaord extention

### PR DESCRIPTION
**Pull request by JWittmeyer**
includes new field, new table & new view for labeling

PRs: 
frontend: https://github.com/code-kern-ai/refinery-ui/pull/55
submodule-model: https://github.com/code-kern-ai/refinery-submodule-model/pull/8
gateway: https://github.com/code-kern-ai/refinery-gateway/pull/43
updater: https://github.com/code-kern-ai/refinery-updater/pull/11
admin-dashboard: https://github.com/code-kern-ai/admin-dashboard/pull/45

### CODE REVIEW

**Comment by jhoetter** [comment]
have the [proxy pr](https://github.com/code-kern-ai/refinery-gateway-proxy/pull/9) in mind, too.

File: controller/labeling_access_link/manager.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    return str(
+        link.heuristic_id
+        if link.link_type == enums.LinkTypes.HEURISTIC.value
+        else link.data_slice_id
+    )
+
+
+def check_link_locked(project_id: str, link_route: str) -> bool:
+
+    if link_route.find("00000000-0000-0000-0000-000000000000") > -1:
```
Comment (jhoetter) [inline controller/labeling_access_link/manager.py]: as this is listed multiple times, I'd suggest to put this into a constant/enum :)

File: controller/project/manager.py [hunk 2]
```diff
@@ -118,3 +123,68 @@ def get_confusion_matrix(
...
+            data_id = session.id
+    else:
+        source_type = enums.LabelSource.MANUAL
+        source_id = None
+        if __no_huddle_id(data_id):
+            data_id = __get_first_data_id(project_id, user_id, huddle_type)
+        if huddle_type == enums.LinkTypes.DATA_SLICE.value:
+            slice_id = data_id
+        elif huddle_type == enums.LinkTypes.HEURISTIC.value:
+            is_data = __get_crowd_label_is_data(project_id, data_id)
```
Comment (jhoetter) [inline controller/project/manager.py]: we need to get away from information source abbreviation `is`. I was totally confused by `is_data` as a non-boolean value

File: controller/record_label_association/manager.py [hunk 3]
```diff
@@ -31,13 +37,68 @@ def is_any_record_manually_labeled(project_id: str):
...
+def get_count_rlas_with_source_id(project_id: str, source_id: str) -> int:
+    return len(
+        record_label_association.get_all_classifications_for_information_source(
+            project_id, source_id
+        )
+    )
+
+
+def __update_annotator_progress(project_id: str, source_id: str, user_id: str):
+    # {"data_slice_id":"a9e54c5a-2b3d-46ba-902d-f6224dd9d24b","annotator_id":"6de83a5b-8e2c-40c2-9923-2b5ed761381b","access_link_id":"c9fcac10-2ba8-40dc-a5ce-ef575dd4a9bf"}
```
Comment (jhoetter) [inline controller/record_label_association/manager.py]: comment

File: graphql_api/types.py [hunk 4]
```diff
@@ -487,6 +488,22 @@ class Meta:
...
+
+    id = graphene.ID(source="id", required=True)
+    name = graphene.String()
+
+    def resolve_name(self, info):
+        if self.link_type == enums.LinkTypes.HEURISTIC.value:
+            return information_source.get(self.project_id, self.heuristic_id).name
+        elif self.link_type == enums.LinkTypes.DATA_SLICE.value:
+            return data_slice.get(self.project_id, self.data_slice_id).name
+        return "Unknown type"
```
Comment (jhoetter) [inline graphql_api/types.py:512]: maybe rather raise an exception?

**Review by jhoetter — APPROVED** [review]
some minor suggestions, but again - nice work :)

File: controller/labeling_access_link/manager.py (hunk 1 above)
Comment (JWittmeyer) [reply controller/labeling_access_link/manager.py]: changed

File: controller/project/manager.py (hunk 2 above)
Comment (JWittmeyer) [reply controller/project/manager.py]: changed to information_source_data & information_source_manager

File: controller/record_label_association/manager.py (hunk 3 above)
Comment (JWittmeyer) [reply controller/record_label_association/manager.py]: removed

File: graphql_api/types.py (hunk 4 above)
Comment (JWittmeyer) [reply graphql_api/types.py:512]: i'd rather not since this "only" a name so having the name "unknown type" wouldn't break the application

**Pull request merged**